src/main/java/dl/news/portal/web/config/oauth2/OAuthSecurityConfig.java

```diff
@@ -0,0 +1,83 @@
+import org.springframework.security.oauth2.client.OAuth2ClientContext;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.security.oauth2.client.filter.OAuth2ClientAuthenticationProcessingFilter;
+import org.springframework.security.oauth2.client.filter.OAuth2ClientContextFilter;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableOAuth2Client;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+import org.springframework.web.filter.CompositeFilter;
+
+import javax.servlet.Filter;
+import java.util.ArrayList;
+import java.util.List;
+
+@Configuration
+@EnableOAuth2Client
+public class OAuthSecurityConfig extends WebSecurityConfigurerAdapter {
+    @Autowired
+    OAuth2ClientContext oauth2ClientContext;
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.
+                authorizeRequests()
+                .antMatchers("/", "/login**", "/webjars/**", "/error**", "/logout**")
+                .permitAll()
+                .mvcMatchers(HttpMethod.GET, "/**").permitAll()
+                .anyRequest().authenticated()
+                .and().logout().logoutSuccessUrl("/").permitAll()
+                .and().csrf().disable()
+                .addFilterBefore(ssoFilter(), BasicAuthenticationFilter.class);
+
+    }
+
+    @Bean
+    @ConfigurationProperties("github")
+    public ClientResources github() {
+        return new ClientResources();
+    }
+
+    private Filter ssoFilter() {
+        CompositeFilter filter = new CompositeFilter();
+        List<Filter> filters = new ArrayList<>();
+        filters.add(ssoFilter(github(), "/login/github"));
+        filter.setFilters(filters);
+        return filter;
+    }
+
+    private Filter ssoFilter(ClientResources client, String path) {
+        OAuth2ClientAuthenticationProcessingFilter filter = new OAuth2ClientAuthenticationProcessingFilter(path);
+        OAuth2RestTemplate template = new OAuth2RestTemplate(client.getClient(), oauth2ClientContext);
+        filter.setRestTemplate(template);
+        UserInfoTokenServices tokenServices = new UserInfoTokenServices(
+                client.getResource().getUserInfoUri(), client.getClient().getClientId());
+        tokenServices.setRestTemplate(template);
+        filter.setTokenServices(tokenServices);
+        filter.setAuthenticationSuccessHandler(oauthAuthenticationSuccessHandler());
+        return filter;
+    }
+
+    @Bean
+    public FilterRegistrationBean<OAuth2ClientContextFilter> oauth2ClientFilterRegistration(OAuth2ClientContextFilter filter) {
+        FilterRegistrationBean<OAuth2ClientContextFilter> registration = new FilterRegistrationBean<>();
+        registration.setFilter(filter);
+        registration.setOrder(-100);
```

Comment:
What does this magic digit mean?

Reply:
Register already available filter with a sufficiently low order that it comes before the main Spring Security filter. In this way  it's should handle redirects signaled by expceptions in authentication requests.